lib/practice_exercises.rb

```diff
@@ -1,13 +1,48 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) as the time taken will depend on the length of list.
+# Space Complexity: I'd like to say O(1) as the space the array takes does not grow. 
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  if list.length == 0 || list.length == 1
+    return list 
+  end 
+
+  i = 0
+  until i == list.length
+    if list[i] == list[i+1]
+      list.delete_at(i)
```

Comment:
`delete_at` causes every item to the right of the deleted item one element to the left.  So `delete_at` has a time complexity of O(n).

This causes your algorithm to have a time complexity of O(n<sup>2</sup>)